src/couch_views/src/couch_views_indexer.erl

```diff
@@ -87,7 +87,32 @@ init() ->
         design_opts => Mrst#mrst.design_opts
     },
 
-    update(Db, Mrst, State).
+    try
+        update(Db, Mrst, State)
+    catch
+        exit:normal ->
+            ok;
+        error:{erlfdb_error, 2101} ->
+            couch_jobs:finish(undefined, Job, Data#{
+                error => transaction_too_large,
+                reason => <<"Transaction exceeds byte limit">>
+            });
+        error:{erlfdb_error, 1031} ->
+            couch_jobs:finish(undefined, Job, Data#{
+                error => transaction_timed_out,
+                reason => <<"Operation aborted because the transaction timed out">>
+            });
+        error:{erlfdb_error, Code} ->
+            couch_jobs:finish(undefined, Job, Data#{
+                error => foundationdb_error,
+                reason => integer_to_binary(Code)
+            });
+        Error:Reason  ->
```

Comment:
Since reason has to be serialized to json, here we might have to translate it to a string / binary.

Try `couch_util:to_binary(Error)` and same for Reason perhaps.

Reply:
maybe it'd be better to encapsulate this "error code to error message" translation into an interface function or even own module within erlfdb for better reuse?

Reply:
@eiri good point. I was thinking of adding an fabric2_fdb_errors file that can handle this.

Reply:
That might good yeah. Erlfdb error module might make more sense. Then we won't have to look it up every time.

Reply:
Added an API function to expose fetching the description from erlfdb directly: https://github.com/cloudant-labs/couchdb-erlfdb/pull/9